app/Models/GeneralAssemblies/GeneralAssembly.php

```diff
@@ -147,7 +148,13 @@ public function isAttended(User $user): bool
      */
     public static function requirementsPassed(User $user): bool
     {
-        $lastAssemblies = GeneralAssembly::all()->sortByDesc('closed_at')->take(2);
+        $year_of_acceptance = $user->educationalInformation->year_of_acceptance;
+        $acceptance_date = now()->setYear($year_of_acceptance)->setMonth(9)->setDay(1);
+        $lastAssemblies = GeneralAssembly::orderBy('closed_at', 'desc')->take(2);
```

Comment:
This returns an empty collection for me... could you check out `GeneralAssembly::all()->sortByDesc('closed_at')->take(2)`? I found it in SemesterEvaluationController, it seems to work.

Reply:
I changed `sortByDesc` to `orderBy` because using the former when I tried to access `$lastAssemblies[1]` (the second last assembly), I found that `$lastAssemblies$` wasn't sorted. (Although the elements seemed to come in order when I tried to iterate it with foreach).

Reply:
Does take execute the query? Just because you have the ->get() call afterwards. Move the ->get to the $lastAssemblies initialiser, in any case.

Reply:
Use Caron::createFromDate($year_of_acceptance, 9, 1)

Reply:
Ps. Use 'orderBy' for query builders, 'sortBy' for collections (after values are queried).